erpnext/regional/india/e_invoice/utils.py

```diff
@@ -1010,13 +1010,29 @@ def bulk_generate_irn(invoices):
 		return failed
 
 	def fetch_and_attach_qrcode_from_irn(self):
-		qrcode = self.get_qrcode_from_irn(self.invoice.irn)
-		if qrcode:
-			qrcode_file = self.create_qr_code_file(qrcode)
-			frappe.db.set_value("Sales Invoice", self.invoice.name, "qrcode_image", qrcode_file.file_url)
-			frappe.msgprint(_("QR Code attached to the invoice"), alert=True)
-		else:
-			frappe.msgprint(_("QR Code not found for the IRN"), alert=True)
+		is_qrcode_file_attached = frappe.db.exists(
+			"File",
+			{
+				"attached_to_doctype": "Sales Invoice",
+				"attached_to_name": self.invoice.name,
+				"file_url": self.invoice.qrcode_image,
+				"attached_to_field": "qrcode_image",
+			},
+		)
+		if not is_qrcode_file_attached:
+			if self.invoice.signed_qr_code:
+				self.attach_qrcode_image(update_url=True)
```

Comment:
```suggestion
				self.attach_qrcode_image()
				frappe.db.set_value("Sales Invoice", self.invoice.name, "qrcode_image", self.invoice.qrcode_image)
```

We can avoid the `kwarg` if it is only used once